python/lsst/obs/base/read_defects.py

```diff
@@ -0,0 +1,40 @@
+from lsst.meas.algorithms import Defects
+import os
+import glob
+from dateutil import parser
+
+
+def read_defects_one_chip(root, chip_name, camera):
+    files = glob.glob(os.path.join(root, chip_name.lower(), '*.ecsv'))
+    parts = os.path.split(root)
+    instrument = os.path.split(parts[0])[1]  # convention is that these reside at <instrument>/defects
+    defect_dict = {}
+    for f in files:
+        date_str = os.path.splitext(os.path.basename(f))[0]
+        valid_start = parser.parse(date_str)
+        defect_dict[valid_start] = Defects.readText(f)
+        check_metadata(defect_dict[valid_start], valid_start, instrument, camera[chip_name].getId())
+    return defect_dict
+
+
+def check_metadata(defects, valid_start, instrument, chip_id):
+    md = defects.getMetadata()
+    finst = md.get('INSTRUME')
+    fchip_id = md.get('DETECTOR')
+    fcalib_date = md.get('CALIBDATE')
+    if not (finst, int(fchip_id), fcalib_date) == (instrument, chip_id, valid_start.isoformat()):
```

Comment:
Note that the failure message will be difficult to parse if any of the metadata is missing. Consider `if None in (finst, fchip_id, fcalib_data): raise...` so the error message says which file is the problem.

Reply:
I want to check for missing metadata as well as differences inferred from the path vs that encoded in the file metadata.  I added a line to print the file it is reading from.  I hope that's ok.

Reply:
I still recommend you include the file name when you raise an exception. I think that will be clearer than printing each file as you go and hoping the user realizes it's the last file printed that gave the problem. But it's your choice. The code you had works.